espr/src/ast/entity.rs

```diff
@@ -190,3 +211,27 @@ pub struct UniqueRule {
     pub name: Option<String>,
     pub attributes: Vec<AttributeDecl>,
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::parser;
+
+    #[test]
+    fn as_supertype_names() {
```

Comment:
This parses `ENTITY A SUPERTYPE OF (B)`, which means `A` is supertype of `B`, i.e. `B` is subtype of `A`. Thus, this should be `as_subtype_names` because it returns `["B"]` for this case and they are subtypes of `self`